src/phoenix/db/migrations/versions/deb2c81c0bb2_dataset_splits.py

```diff
@@ -0,0 +1,130 @@
+"""dataset_splits
+
+Revision ID: deb2c81c0bb2
+Revises: 58228d933c91
+Create Date: 2025-09-08 15:50:12.066217
+
+"""
+
+from typing import Any, Sequence, Union
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy import JSON
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.ext.compiler import compiles
+
+
+class JSONB(JSON):
+    # See https://docs.sqlalchemy.org/en/20/core/custom_types.html
+    __visit_name__ = "JSONB"
+
+
+@compiles(JSONB, "sqlite")
+def _(*args: Any, **kwargs: Any) -> str:
+    # See https://docs.sqlalchemy.org/en/20/core/custom_types.html
+    return "JSONB"
+
+
+JSON_ = (
+    JSON()
+    .with_variant(
+        postgresql.JSONB(),
+        "postgresql",
+    )
+    .with_variant(
+        JSONB(),
+        "sqlite",
+    )
+)
+
+_Integer = sa.Integer().with_variant(
+    sa.BigInteger(),
+    "postgresql",
+)
+
+
+# revision identifiers, used by Alembic.
+revision: str = "deb2c81c0bb2"
+down_revision: Union[str, None] = "58228d933c91"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # Create dataset_splits table
+    op.create_table(
+        "dataset_splits",
+        sa.Column("id", _Integer, primary_key=True),
+        sa.Column("name", sa.String, nullable=False, unique=True, index=True),
```

Comment:
use constraint instead of index
```suggestion
        sa.Column("name", sa.String, nullable=False, unique=True),
```